aya/src/features.rs

```diff
@@ -0,0 +1,182 @@
+// //! A collection of ebpf feature helpers that can determine kernel capabilities.
+// //!
+// //! Basing kernel capabilities on kernel version is not sufficient since some
+// //! distros will backport ebpf functionality to older kernels.
+// //!
+
+// use std::mem::size_of;
+
+// use aya_obj::{generated::bpf_map_type, maps::LegacyMap, EbpfSectionKind, Map};
+// use thiserror::Error;
+
+// use crate::maps::{MapData, MapError};
+
+// /// An error ocurred working with a pinned BPF object.
+// #[derive(Error, Debug)]
+// pub enum FeatureError {
+//     /// An error ocurred making a syscall.
+//     #[error(transparent)]
+//     MapError(#[from] MapError),
+// }
+
+use crate::sys::is_ringbuf_supported as ringbuf_supported;
+
+pub fn is_ringbuf_supported() -> bool {
+    ringbuf_supported()
+}
+
+// /// Result type used for the feature helpers
+// pub type Result<T> = std::result::Result<T, FeatureError>;
+
+// fn probe_map_create(map_type: bpf_map_type) -> Result<bool> {
```

Comment:
I actually like the idea of having a separate `features.rs` file. Maybe you could file a prerequisite PR, which moves all `is_*_supported` functions from `bpf.rs` to `features.rs`?

I would be also fine with adding `is_ringbuf_supported` to `bpf.rs` and doing this refactor later.

But either way, let's do it without adding commented code. :)

Reply:
Was actually a mistake making that 2nd commit. I'll probably close this PR since the scope has changed (see chat) and create a new when I have some working code that's in line with the new proposed solution and your comment